pkg/chartRepo/ChartRepositoryService.go

```diff
@@ -99,22 +100,38 @@ func NewChartRepositoryServiceImpl(logger *zap.SugaredLogger, repoRepository cha
 }
 
 // Private helm charts credentials are saved as secrets
-func (impl *ChartRepositoryServiceImpl) CreateSecretDataForPrivateHelmChart(request *ChartRepoDto) (secretData map[string]string) {
+func (impl *ChartRepositoryServiceImpl) CreateSecretDataForHelmChart(request *ChartRepoDto, isPrivateChart bool) (secretData map[string]string) {
 	secretData = make(map[string]string)
-	secretData[NAME] = fmt.Sprintf("%s-%s", request.Name, uuid.New().String()) // making repo name unique so that "helm repp add" command in argo-repo-server doesn't give error
-	secretData[USERNAME] = request.UserName
-	secretData[PASSWORD] = request.Password
+	secretData[NAME] = fmt.Sprintf("%s-%s", request.Name, uuid.New().String()) // making repo name unique so that "helm repo add" command in argo-repo-server doesn't give error
 	secretData[TYPE] = HELM
 	secretData[URL] = request.Url
-	isInsecureConnection := "true"
-	if !request.AllowInsecureConnection {
-		isInsecureConnection = "false"
+	if isPrivateChart {
+		secretData[USERNAME] = request.UserName
+		secretData[PASSWORD] = request.Password
+		isInsecureConnection := "true"
+		if !request.AllowInsecureConnection {
+			isInsecureConnection = "false"
+		}
+		secretData[INSECRUE] = isInsecureConnection
 	}
-	secretData[INSECRUE] = isInsecureConnection
+
 	return secretData
 }
 
 func (impl *ChartRepositoryServiceImpl) CreateChartRepo(request *ChartRepoDto) (*chartRepoRepository.ChartRepo, error) {
+	allChartsRepos, err := impl.repoRepository.FindAll()
+	if err != nil {
+		impl.logger.Errorw("error in getting list of all chart repos", "err", err)
+		return nil, err
+	}
+	if len(allChartsRepos) == 0 {
+		return nil, errors.New("no chart repo found in db")
```

Comment:
is it an error or expected condition ?

Reply:
it's just handling an a corner case when error is nil and if somehow there are no entries in db for chart repos

Reply:
have handled this case differently, if no data is present will return nil and validation result only, and then no need to trigger chart sync